From e09aeb76cba4ad48b3d753f58476c0b048495148 Mon Sep 17 00:00:00 2001
From: Alexander Gattin <alexander_gattin@xyratex.com>
Date: Sun, 31 Mar 2013 20:51:55 +0300
Subject: [PATCH 09/15] db/pair: don't break m0_db_pair_invariant() in
 pair_prep()

When freeing key/rec->data and setting it to NULL,
also reset its ->size to zero, to avoid breaking
m0_db_pair_invariant()
---
 db/db.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/db/db.c b/db/db.c
index d402856..5b7d60e 100644
--- a/db/db.c
+++ b/db/db.c
@@ -635,10 +635,12 @@ static void pair_prep(struct m0_db_pair *pair)
 	if (pair->dp_key.db_type == DBT_ALLOC) {
 		m0_free(key->data);
 		key->data = NULL;
+		key->size = 0;
 	}
 	if (pair->dp_rec.db_type == DBT_ALLOC) {
 		m0_free(rec->data);
 		rec->data = NULL;
+		rec->size = 0;
 	}
 }
 
-- 
1.8.3.2

